plugins/module_utils/network/sonic/config/network_policy/network_policy.py

```diff
@@ -0,0 +1,392 @@
+                        policy_dict['applications'] = {'application': app_list}
+                if policy_dict:
+                    policy_list.append(policy_dict)
+            if policy_list:
+                payload = {'openconfig-network-policy-ext:network-policies': {'network-policy': policy_list}}
+                request = {'path': NETWORK_POLICY_PATH, 'method': PATCH, 'data': payload}
+
+        return request
+
+    def get_delete_network_policy_requests(self, commands, is_delete_all):
+        requests = []
+
+        if not commands:
+            return requests
+        if is_delete_all:
+            requests.append({'path': NETWORK_POLICY_PATH, 'method': DELETE})
+            return requests
+
+        for policy in commands:
+            number = policy.get('number')
+            applications = policy.get('applications')
+
+            if number and not applications:
+                requests.append(self.get_delete_network_policy_request(number))
+                continue
+            if applications:
+                for app in applications:
+                    app_type = app.get('app_type')
+                    vlan_id = app.get('vlan_id')
+                    tagged = app.get('tagged')
+                    priority = app.get('priority')
+                    dscp = app.get('dscp')
+
+                    if app_type and len(app) == 1:
+                        requests.append(self.get_delete_network_policy_request(number, app_type))
+                        continue
+                    if vlan_id:
+                        requests.append(self.get_delete_network_policy_request(number, app_type, 'vlan-id'))
+                    if tagged is not None:
+                        requests.append(self.get_delete_network_policy_request(number, app_type, 'tagged'))
+                    if priority is not None:
+                        requests.append(self.get_delete_network_policy_request(number, app_type, 'priority'))
+                    if dscp is not None:
+                        requests.append(self.get_delete_network_policy_request(number, app_type, 'dscp'))
+
+        return requests
+
+    def get_delete_network_policy_request(self, number, app_type=None, attr=None):
+        url = '%s/network-policy=%s' % (NETWORK_POLICY_PATH, number)
+        if app_type:
+            url += '/applications/application=%s' % (app_type.upper())
+        if attr:
+            url += '/config/%s' % (attr)
+        request = {'path': url, 'method': DELETE}
+        return request
+
+    def get_replaced_config(self, want, have):
+        config_list = []
+        requests = []
+
+        if not want or not have:
+            return config_list, requests
+
+        policy_dict = {policy.get('number'): policy for policy in have}
+        for policy in want:
+            number = policy.get('number')
+            cfg_policy = policy_dict.get(number)
+
+            if not cfg_policy:
+                continue
+            if policy != cfg_policy:
```

Comment:
Since a policy contains a list of applications, this equality check would return false even when the contents are the same, but the order of the list is different.
Please ensure the contents are ordered prior to this check.